client/components/disputed-order-notice/index.js

```diff
@@ -0,0 +1,210 @@
+	if ( ! now.isBefore( dueBy ) ) {
+		return null;
+	}
+
+	return (
+		<DisputeNeedsResponseNotice
+			chargeId={ chargeId }
+			disputeReason={ dispute.reason }
+			formattedAmount={ formatExplicitCurrency(
+				dispute.amount,
+				dispute.currency
+			) }
+			isPreDisputeInquiry={ isInquiry( dispute ) }
+			dueBy={ dueBy }
+			countdownDays={ Math.floor( dueBy.diff( now, 'days', true ) ) }
+			disputeDetailsUrl={ disputeDetailsUrl }
+		/>
+	);
+};
+
+const DisputeNeedsResponseNotice = ( {
+	disputeReason,
+	formattedAmount,
+	isPreDisputeInquiry,
+	dueBy,
+	countdownDays,
+	disputeDetailsUrl,
+} ) => {
+	useEffect( () => {
+		wcpayTracks.recordEvent( 'wcpay_order_dispute_notice_view', {
+			is_inquiry: isPreDisputeInquiry,
+			dispute_reason: disputeReason,
+			due_by_days: countdownDays,
+		} );
+	}, [ isPreDisputeInquiry, disputeReason, countdownDays ] );
+
+	const titleStrings = {
+		// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+		dispute_default: __(
+			// eslint-disable-next-line max-len
+			'This order has been disputed in the amount of %1$s. The customer provided the following reason: %2$s. Please respond to this dispute before %3$s.',
+			'woocommerce-payments'
+		),
+		// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+		inquiry_default: __(
+			// eslint-disable-next-line max-len
+			'The card network involved in this order has opened an inquiry into the transaction with the following reason: %2$s. Please respond to this inquiry before %3$s, just like you would for a formal dispute.',
+			'woocommerce-payments'
+		),
+		// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+		dispute_urgent: __(
+			'Please resolve the dispute on this order for %1$s labeled "%2$s" by %3$s.',
+			'woocommerce-payments'
+		),
+		// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+		inquiry_urgent: __(
+			'Please resolve the inquiry on this order for %1$s labeled "%2$s" by %3$s.',
+			'woocommerce-payments'
+		),
+	};
+
+	let buttonLabel = __( 'Respond now', 'woocommerce-payments' );
+	let suffix = '';
+
+	let titleText = isPreDisputeInquiry
+		? titleStrings.inquiry_default
+		: titleStrings.dispute_default;
+
+	// If the dispute is due within 7 days, adjust wording and highlight urgency.
+	if ( countdownDays < 7 ) {
+		titleText = isPreDisputeInquiry
+			? titleStrings.inquiry_urgent
+			: titleStrings.dispute_urgent;
+
+		suffix = sprintf(
+			// Translators: %s is the number of days left to respond to the dispute.
+			_n(
+				'(%s day left)',
+				'(%s days left)',
+				countdownDays,
+				'woocommerce-payments'
+			),
+			countdownDays
+		);
+	}
+
+	const title = sprintf(
+		titleText,
+		formattedAmount,
+		reasons[ disputeReason ].display,
+		dateI18n( 'M j, Y', dueBy.local().toISOString() )
+	);
```

Comment:
The logic for generating the copy is still a little convoluted and complex, but leaving as is since that code is well tested, keep the refactor contained to the most pressing issue (manage side effects).